### PR TITLE
Improve PostgreSQL INSERT performance

### PR DESCRIPTION
**Pull request by wendigo**
Extracted from https://github.com/trinodb/trino/pull/12400

### CODE REVIEW

**Comment by wendigo** [comment]
```
reWriteBatchedInserts = bool Enable optimization to rewrite and collapse compatible INSERT statements that are batched. If enabled, pgjdbc rewrites batch of insert into ... values(?, ?) into insert into ... values(?, ?), (?, ?), ... That reduces per-statement overhead. The drawback is if one of the statements fail, the whole batch fails. The default value is false. The option is available since 9.4.1208
```

https://jdbc.postgresql.org/documentation/94/connect.html

**Review by kokosing — APPROVED** [review]

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java [hunk 1]
```diff
@@ -56,6 +58,8 @@ public void setup(Binder binder)
     @ForBaseJdbc
     public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
     {
-        return new DriverConnectionFactory(new Driver(), config, credentialProvider);
+        Properties connectionProperties = new Properties();
+        connectionProperties.put("reWriteBatchedInserts", "true");
```
Comment (electrum) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:62]: Use `org.postgresql.PGProperty.REWRITE_BATCHED_INSERTS.getName()`

Comment (electrum) [reply plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:62]: These are case sensitive so using the property enum avoids mistakes

**Review by electrum — APPROVED** [review]

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java [hunk 2]
```diff
@@ -56,6 +60,8 @@ public void setup(Binder binder)
     @ForBaseJdbc
     public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
     {
-        return new DriverConnectionFactory(new Driver(), config, credentialProvider);
+        Properties connectionProperties = new Properties();
+        connectionProperties.put(REWRITE_BATCHED_INSERTS.name(), TRUE.toString());
```
Comment (electrum) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:64]: We can use `"true"` here

Comment (electrum) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:64]:
```suggestion
        connectionProperties.put(REWRITE_BATCHED_INSERTS.name(), "true");
```

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java [hunk 3]
```diff
@@ -61,7 +60,7 @@ public void setup(Binder binder)
     public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
     {
         Properties connectionProperties = new Properties();
-        connectionProperties.put(REWRITE_BATCHED_INSERTS.name(), TRUE.toString());
+        connectionProperties.put(REWRITE_BATCHED_INSERTS.name(), "true");
```
Comment (electrum) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:63]: This needs to be `getName()`

File: plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java [hunk 4]
```diff
@@ -56,6 +59,8 @@ public void setup(Binder binder)
     @ForBaseJdbc
     public ConnectionFactory getConnectionFactory(BaseJdbcConfig config, CredentialProvider credentialProvider)
     {
-        return new DriverConnectionFactory(new Driver(), config, credentialProvider);
+        Properties connectionProperties = new Properties();
+        connectionProperties.put(REWRITE_BATCHED_INSERTS.name(), "true");
```
Comment (electrum) [inline plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClientModule.java:63]:
```suggestion
        connectionProperties.put(REWRITE_BATCHED_INSERTS.getName(), "true");
```

**Review by hashhar — APPROVED** [review]

**Pull request merged**